src/CloudNative.CloudEvents/CloudEventAttributeType.cs

```diff
@@ -51,6 +51,52 @@ public abstract class CloudEventAttributeType
         /// </summary>
         public static CloudEventAttributeType Timestamp { get; } = new TimestampType();
 
+        /// <summary>
+        /// Enum of event types, to allow efficient switching over CloudEventAttributeType.
+        /// Each attribute type has a unique value, returned by <see cref="OrdinalType"/>.
+        /// </summary>
+        /// <remarks>
+        /// This type is nested as relatively few consumers will need to use it.
+        /// </remarks>
+        public enum Ordinal
```

Comment:
Agree with the idea of moving this to the Core namespace rather than nesting it here.

Reply:
Yup - I've got another PR *nearly* ready for this... and which avoids having a public property on the class as well. (It's an internal property which you get at via a static method in a static class in the Core namespace - basically that extra layer of indirection effectively obscures it from "regular" users.)